src/plugins/data/public/search/search_interceptor.ts

```diff
@@ -173,16 +176,16 @@ export class SearchInterceptor {
    * error notification per session.
    * @internal
    */
-  private showTimeoutError = debounce(
-    (e: SearchTimeoutError) => {
+  private showTimeoutError = (e: SearchTimeoutError, sessionId?: string) => {
+    if (!sessionId || !this.deps.session.getSessionTimeoutNotified()) {
+      this.deps.session.setSessionTimeoutNotified();
+
       this.deps.toasts.addDanger({
         title: 'Timed out',
         text: toMountPoint(e.getErrorMessage(this.application)),
       });
-    },
-    30000,
-    { leading: true, trailing: false }
-  );
+    }
+  };
```

Comment:
Thoughts about using `memoize` from lodash here (and just using the `sessionId` as the resolver)? It removes the need for any of the `sessionTimeoutNotified` logic in this PR and simplifies things a bit.

Reply:
Great idea!